src/components/attack.py

```diff
@@ -11,46 +11,47 @@ def __init__(self, focus, *groups) -> None:
         self.horizontal_surface = pg.Surface([self.attack_width, self.attack_height])
         self.vertical_surface = pg.Surface([self.attack_height, self.attack_width])
         self.attack_sequence = 0  # Track the current attack sequence stage
-        self.last_attack_time = 0  # Time since the last attack input
-        self.sequence_timeout = 50000  # Time in milliseconds before the sequence resets
         self.attack_sprite = AttackSprite(focus, *groups)
+        self.attack_timeout_timer = Timer(threshold=5) # amount of seconds before times out
+
+
 
     def handle_attack_input(self, groups : pg.sprite.Group):
-        current_time = pg.time.get_ticks()
         if self.attack_sequence == 0:
             self.perform_attack(groups)
             self.attack_sequence = 1
-            self.last_attack_time = current_time
+            self.attack_timeout_timer.reset()
             print("Performed first attack.")
 
         elif self.attack_sequence == 1:
-            if current_time - self.last_attack_time < self.sequence_timeout:
-                self.perform_attack(groups)
-                self.attack_sequence = 2
-                self.last_attack_time = current_time
-                print("Performed second attack.")
+            self.perform_attack(groups)
+            self.attack_sequence = 2
+            self.attack_timeout_timer.reset()
+            print("Performed second attack.")
 
-            else:
-                self.reset_sequence()
         elif self.attack_sequence == 2:
-            if current_time - self.last_attack_time < self.sequence_timeout:
-                self.attack_sprite.perform_smash_attack(groups)
-                self.reset_sequence()
-            else:
-                self.reset_sequence()
-                
-    def reset_sequence(self):
-        self.attack_sequence = 0
-        self.last_attack_time = 0
-        print("Attack sequence reset.")
+            self.attack_sprite.perform_smash_attack(groups)
```

Comment:
Is there any reason that perform_smash_attack is in tongue attack function, and not this attack sequence one? (if there isnt any reason, could you swap it over before merging :3 )